lib/auth/clusterconfig/clusterconfigv1/service.go

```diff
@@ -453,3 +458,168 @@ func (s *Service) ResetClusterNetworkingConfig(ctx context.Context, _ *clusterco
 
 	return nil, trace.LimitExceeded("failed to reset networking config within %v iterations", iterationLimit)
 }
+
+// GetSessionRecordingConfig returns the locally cached networking configuration.
+func (s *Service) GetSessionRecordingConfig(ctx context.Context, _ *clusterconfigpb.GetSessionRecordingConfigRequest) (*types.SessionRecordingConfigV2, error) {
+	authzCtx, err := s.authorizer.Authorize(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if err := authzCtx.CheckAccessToKind(types.KindSessionRecordingConfig, types.VerbRead); err != nil {
+		if err2 := authzCtx.CheckAccessToKind(types.KindClusterConfig, types.VerbRead); err2 != nil {
+			return nil, trace.Wrap(err)
+		}
+	}
+
+	netConfig, err := s.cache.GetSessionRecordingConfig(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	cfgV2, ok := netConfig.(*types.SessionRecordingConfigV2)
+	if !ok {
+		return nil, trace.Wrap(trace.BadParameter("unexpected session recording config type %T (expected %T)", netConfig, cfgV2))
+	}
+	return cfgV2, nil
+}
+
+// CreateSessionRecordingConfig creates a new cluster networking configuration if one does not exist.
+// This is an internal API and is not exposed via [clusterconfigv1.ClusterConfigServiceServer]. It
+// is only meant to be called directly from the first time an Auth instance is started.
+func (s *Service) CreateSessionRecordingConfig(ctx context.Context, cfg types.SessionRecordingConfig) (types.SessionRecordingConfig, error) {
+	authzCtx, err := s.authorizer.Authorize(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if !authz.HasBuiltinRole(*authzCtx, string(types.RoleAuth)) {
+		return nil, trace.AccessDenied("this request can be only executed by an auth server")
+	}
+
+	created, err := s.backend.CreateSessionRecordingConfig(ctx, cfg)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	cfgV2, ok := created.(*types.SessionRecordingConfigV2)
+	if !ok {
+		return nil, trace.Wrap(trace.BadParameter("unexpected session recording config type %T (expected %T)", created, cfgV2))
+	}
+
+	return cfgV2, nil
+}
+
+// UpdateSessionRecordingConfig conditionally updates an existing networking configuration if the
+// value wasn't concurrently modified.
+func (s *Service) UpdateSessionRecordingConfig(ctx context.Context, req *clusterconfigpb.UpdateSessionRecordingConfigRequest) (*types.SessionRecordingConfigV2, error) {
```

Comment:
Should `UpdateSessionRecordingConfig` and `Upsert..` also set "dynamic" origin? If not then why only the auth preference?